meinberlin/apps/dashboard/views.py

```diff
@@ -205,6 +205,23 @@ class ModuleDeleteView(PermissionRequiredMixin,
     success_message = _('The module has been deleted')
 
     def delete(self, request, *args, **kwargs):
+        module = self.get_object()
+        # only unpublished modules can be deleted
+        if not module.is_draft:
+            messages.error(request,
+                           _('Module added to a project cannot be '
+                             'deleted.'))
+            # try to stay on same page
+            if 'referrer' in request.POST:
+                redirect = request.POST['referrer']
+            elif 'HTTP_REFERER' in self.request.META:
+                redirect = request.META['HTTP_REFERER']
+            else:
+                redirect = reverse('a4dashboard:project-edit', kwargs={
+                    'project_slug': module.project.slug
```

Comment:
I am a bit unhappy about putting all of that here. But the referrer stuff in the success_url is a bit different, because we cannot refer back to the module forms, when the module was deleted. And as long as we only have one thing to test in the delete, maybe it's fine to keep it here?